open-craft/opencraft
# [BB-3585] Add initial setup for e2e tests

This adds the base for e2e test in OCIM with react frontend

**JIRA Tickets**: [BB-3585](https://tasks.opencraft.com/browse/BB-3585)

**Testing Instruction**
1. Checkout to this branch
2. Run `vagrant up` and `vagrant ssh` to checkin into the devstack.
3. Run `make requirements`
4. Run `./bin/install-supported-firefox`
4. Run `make test.e2e`
5. Test that the frontend is build and e2e tests are run and passing.
6. Check that the Circle CI build `e2e-tests` is passing.

**Author Notes**

1. selenium and firefox need to be updated to `3.141.0`(added in requirements) and `85+` (updated in `./bin/install-supported-firefox`) respectively
2. We have `registration.tests.utils.BrowserTestMixin` using `selenium` but I could not find its use anywhere.
